plugins/BEdita/Core/src/Model/Behavior/SearchableBehavior.php

```diff
@@ -121,78 +133,50 @@ public function findQuery(Query $query, array $options)
             ]);
         }
 
-        $minLength = $this->getConfig('minLength');
-        $maxWords = $this->getConfig('maxWords');
-        $words = [$options['string']];
-        if (filter_var($options['exact'], FILTER_VALIDATE_BOOLEAN) !== true) {
-            $words = preg_split('/\W+/', $options['string']); // Split words.
-        }
-        $words = array_unique(array_map( // Escape `%` and `\` characters in words.
-            function ($word) {
-                return str_replace(
-                    ['%', '\\'],
-                    ['\\%', '\\\\'],
-                    mb_strtolower($word)
-                );
-            },
-            array_filter( // Filter out words that are too short.
-                $words,
-                function ($word) use ($minLength) {
-                    return mb_strlen($word) >= $minLength;
-                }
-            )
-        ));
-        if (count($words) === 0) {
-            // Query contained only short words.
-            throw new BadFilterException([
-                'title' => __d('bedita', 'Invalid data'),
-                'detail' => __d('bedita', 'query strings must be at least {0} characters long', $minLength),
-            ]);
-        }
-        if ($maxWords > 0 && count($words) > $maxWords) {
-            // Conditions with too many words would make our database hang for a long time.
-            throw new BadFilterException([
-                'title' => __d('bedita', 'Invalid data'),
-                'detail' => 'query string too long',
-            ]);
-        }
+        unset($options[0], $options['string']);
 
-        // Concat all fields into a single, lower-cased string.
-        $fields = [];
-        /** @var \Cake\ORM\Table $table */
-        $table = $query->getRepository();
-        foreach (array_keys($this->getFields()) as $field) {
-            $fields[] = $query->func()->coalesce([
-                $table->aliasField($field) => 'identifier',
-                '',
-            ]);
-            $fields[] = ' '; // Add a spacer.
+        return $this->getAdapter(Configure::read('Search.use', 'default'))
+            ->search($query, $text, $options);
+    }
+
+    /**
+     * Fit configuration of `SimpleAdapter` to maintain backward compatibility with 5.x.
+     *
+     * @param \BEdita\Core\Search\Adapter\SimpleAdapter $adapter The adapter.
+     * @return void
+     */
+    protected function fitSimpleAdapterConf(SimpleAdapter $adapter): void
+    {
+        $config = array_intersect_key(
+            $this->getConfig(),
+            array_flip(['minLength', 'maxWords'])
+        );
+        $adapter->setConfig($config);
+
+        // Config keys that must be overridden
+        foreach (['columnTypes', 'fields'] as $key) {
+            $conf = $this->getConfig($key);
+            if (!is_array($conf)) {
+                continue;
+            }
+
+            // `fields` key in SimpleAdapter is changed.
```

Comment:
Maybe we could raise a `deprecationWarning('blahblahblah')` when we reach this point?